pkg/controllers/resources/pods/syncer.go

```diff
@@ -70,9 +77,20 @@ func New(ctx *synccontext.RegisterContext) (syncer.Object, error) {
 
 		podTranslator: podTranslator,
 		nodeSelector:  nodeSelector,
+		tolerations:   tolerations,
 	}, nil
 }
 
+func parseToleration(toleration string) (key string, value string, effect string) {
```

Comment:
Currently this feels very static, could we refactor this into:

I see the difference between taints and tolerations, but can we not have a similar syntax for operations to how label expressions work with equal and exists:
- key1=value1:Effect => Operator Equals
- key1:Effect => Operator Exists

Also it would be good to support the case where effect is empty as this will match all taint effects:
- key1=value1 
- key1

There is also another special case and this just the operator `Exists` without effect or key that would match every taint, we could express this via:
- `*`

Reply:
Oh that's a great idea. It only becomes a problem if they introduce a new operator like NoExists or something like that. But with what exists currently, no worries.

I may get a lot of inspiration from https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/taint/utils.go#L77-L118 then. It does almost everything I need logic wise and I could try and import some of the validation function they use even.

And add something separate for the operator.


> There is also a special case where just the operator Exists without effect or key can be specified, which would match every taint, we could express this via:

Oh I never thought about that, that makes a lot of sense.

Lastly, seems like that parsing function could be
1) reused for both the cmd side and the syncer
2) be a good candidate to have a good suite of unit tests given the logic involved

Should I move this to utils? Or is there a better place for this?

Reply:
@kuuji sounds great, moving that to utils is a good idea!